strawberryfields/apps/graph/utils.py

```diff
@@ -153,3 +153,76 @@ def is_subgraph(subgraph: Iterable, graph: nx.Graph):
         return set(subgraph).issubset(graph.nodes)
     except TypeError:
         raise TypeError("subgraph and graph.nodes must be iterable")
+
+
+def is_clique(graph: nx.Graph) -> bool:
+    """Determines if the input graph is a clique
+
+    Args:
+        graph (nx.Graph): the input graph
+
+    Returns:
+        bool: returns ``True`` if input graph is a clique and ``False`` otherwise
+    """
+    edges = graph.edges
+    nodes = graph.order()
+
+    return len(edges) == nodes * (nodes - 1) / 2  # a clique has of n nodes has n*(n-1)/2 edges
+
+
+def c_0(subgraph: list, graph: nx.Graph):
```

Comment:
Still not sure on whether to rename `subgraph` to `clique` :thinking:

Reply:
This is slightly anecdotal, but `subgraph` to me sounds like it should be a Graph object, or perhaps related to `graph` arguments/`graph_type`. Since this is instead a list, I would definitely choose `clique` over `subgraph`.

Reply:
Agree, `clique` is good here because you explicitly reject anything that's not. In the case of `is_clique` I would prefer `subgraph`.